Find all matching SDP codecs for defaults

tylergibson opened the pull request:
Small fix for setting H264 and other multi-profile codecs as default for SDP offers.

KaptenJansson commented on src/web_app/js/sdputils.js:
```diff
@@ -330,9 +330,15 @@ function maybePreferCodec(sdp, type, dir, codec) {
+  var payload = null;
+  for (var i = 0; i < sdpLines.length; i++) {
+    var index = findLineInRange(sdpLines, i, -1, 'a=rtpmap', codec);
+    if(index !== null) {
```
Space between `if` and `(`.

tylergibson commented:
@KaptenJansson Thanks, fixed the spacing nit and amended the commit.

Also I believe this resolves #492, as it prioritizes all h264 profiles and will cycle offer candidates until one is accepted.

jianjunz commented on src/web_app/js/sdputils.js:
```diff
@@ -330,9 +330,15 @@ function maybePreferCodec(sdp, type, dir, codec) {
+      payload = getCodecPayloadTypeFromLine(sdpLines[index]);
+      if (payload) {
+        sdpLines[mLineIndex] = setDefaultCodec(sdpLines[mLineIndex], payload);
+      }
```
SDP lines from ```i``` to ```index``` was scanned in ```findLineInRange```. Perhaps we can skip them by adding ```i = index;``` here.

tylergibson replied:
@jianjunz I originally had exactly that, but removed it thinking it was redundant.  You're right though it would skip the next index - i iterations. I'll update

tylergibson commented:
Thanks to @jianjunz, added the index to i assignment to reduce loop iterations.  Also added null break to exit the search once no match is found.

Finally, updated findLineInRange to search ascending or descending.  By doing a descending search we end up with all matched codecs in the same preference order as the original SDP, preserving client preference.

tylergibson commented:
@KaptenJansson Pingback, anything blocking accepting this PR?

KaptenJansson commented:
Will publish a new AppRTC release later this week.
